core/src/main/java/io/cucumber/core/plugin/MessageFormatter.java

```diff
@@ -12,8 +12,7 @@
 public final class MessageFormatter implements ConcurrentEventListener {
 
     private final Writer writer;
-    private final JsonFormat.Printer jsonPrinter = JsonFormat.printer()
-            .omittingInsignificantWhitespace();
+    private final ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
`ObjectMapper` outputs without whitespace or newlines by default.

Reply:
There is a `JSON` object in `messasges` that wraps the `ObjectMapper` with the right settings.

Reply:
Fixed, thanks